modules/objdetect/src/face_recognize.cpp

```diff
@@ -0,0 +1,186 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+
+#include "opencv2/imgproc.hpp"
+#include "opencv2/core.hpp"
+#include "opencv2/dnn.hpp"
+
+#include <algorithm>
+
+namespace cv
+{
+
+class DNNFaceRecognizerImpl : public DNNFaceRecognizer
+{
+public:
+    DNNFaceRecognizerImpl(const String& onnx_path)
+    {
+        this->model = dnn::readNet(onnx_path);
+    };
+    void AlignCrop(InputArray _src_img, InputArray _face_mat, OutputArray _aligned_img) const override
+    {
+        Mat face_mat = _face_mat.getMat();
+        float src_point[5][2];
+        for (int row = 0; row < 5; ++row)
+        {
+            for(int col = 0; col < 2; ++col)
+            {
+                src_point[row][col] = face_mat.at<double>(1, row*2+col+4);
+            }
+        }
+        Mat warp_mat = getSimilarityTransformMatrix(src_point);
+        warpAffine(_src_img, _aligned_img, warp_mat, Size(112, 112), INTER_LINEAR);
+    };
+    void facefeature(InputArray _aligned_img, OutputArray _face_feature) override
+    {
+        Mat inputBolb = dnn::blobFromImage(_aligned_img, 1, Size(112, 112), Scalar(0, 0, 0), true, false);
+        this->model.setInput(inputBolb);
+        return this->model.forward(_face_feature);
```

Comment:
imho it needs a clone() here, else the 2nd run of the network overwrites the 1st (shallow copy of the network's last layer)
and both features are the same

Reply:
Sorry for the late response. I deleted the '.clone()' because I think maintaining the extracted feature is up to the user. User may allow network to overwrite the last feature to reduce excess memory usage.

Reply:
Sorry for the late response. I deleted the '.clone()' because I think maintaining the extracted feature is up to the user. User may allow network to overwrite the last feature to reduce excess memory usage.
And I'm sorry for another mistake, this function need not return any value.

Maybe, I can delete 'return', and add '.clone()' into the sample code.
```cpp
faceRecognizer->facefeature(aligned_face1, feature1);
feature1 = feature1.clone();
faceRecognizer->facefeature(aligned_face2, feature2);
feature2 = feature2.clone();
```